### PR TITLE
Support stringifying non-exhaustive enum to json

### PR DESCRIPTION
**Pull request by pfgithub**
```zig
test "stringify non-exhaustive enum" {
    const E = enum(u8) {
        foo = 0,
        _,
    };
    try testStringify("\"foo\"", E.foo, .{});
    try testStringify("1", @as(E, @enumFromInt(1)), .{});
}
```

It stringifies the named ones to strings and unnamed ones to a number, which parses just fine into an enum already.

I'm not sure if there's a better way to detect if the active field is unnamed - `if` doesn't support both `_` and `else`, and tagName doesn't return an optional for non-exhaustive enums.

### CODE REVIEW

File: lib/std/json/stringify.zig [hunk 1]
```diff
@@ -540,11 +540,26 @@ pub fn WriteStream(
...
 
+                    if (!enum_info.is_exhaustive) {
+                        inline for (enum_info.fields) |field| {
+                            if (value == @field(T, field.name)) {
+                                break;
+                            }
+                        } else {
+                            try self.valueStart();
+                            try self.stream.print("{d}", .{@intFromEnum(value)});
+                            return self.valueDone();
```
Comment (thejoshwolfe) [inline lib/std/json/stringify.zig:556]: there's a precedent to unwrap a type and then recurse for the underlying type like this:
```suggestion
                            return self.write(@intFromEnum(value));
```

**Review by thejoshwolfe — CHANGES_REQUESTED** [review]
great idea! and thanks for the test! just missing documentation. The other suggestion is optional.

**Comment by pfgithub** [comment]
Added to the documentation comment, switched to recurse instead of writing raw text, and updated for the lowercase std.builtin.Type change

**Review by thejoshwolfe — APPROVED** [review]
thanks!

**Pull request merged**